applicationset/controllers/applicationset_controller.go

```diff
@@ -1287,13 +1288,22 @@ func (r *ApplicationSetReconciler) migrateStatus(ctx context.Context, appset *ar
 	return nil
 }
 
+func keys[T any](m map[string]T) []string {
```

Comment:
the `sort` is really what is important in this function, either rename the function to `sortedKeys` or move the `sort.Strings(keys)` outside the function.

```
sort.Slice(statuses, func(i, j int) bool {
	return statuses[i].Name < statuses[j].Name
})
```